design/keycloak.x/configuration.md

```diff
@@ -563,6 +563,89 @@ Examples of commands are:
 Commands may support additional options to change how they are executed and behavior. For that, commands should provide the
 help and usage messages to document how they can be used as well as the different options they support. 
 
+### The `provider` command
+
+The `provider` command provides utilities to query, (un)install, and enable/disable providers. The usage of this command is the following:
+
+```
+kc.[sh|bat] spi [list-spi|list] [
+```
+
+By running `kc.[sh|bat] provider`, the CLI should list all installed providers:
+
+```
+Listing installed providers (see --help for more options):
+
+hostname
+        Default: default
+        Provider (name: default, factory: class org.keycloak.url.DefaultHostnameProviderFactory)
+
+localeSelector
+        Default: default
+        Provider (name: default, factory: class org.keycloak.locale.DefaultLocaleSelectorProviderFactory)
+
+localeUpdater
+        Default: default
+        Provider (name: default, factory: class org.keycloak.locale.DefaultLocaleUpdaterProviderFactory)
+
+To get more information, including the complete list of providers for a SPI, append `--full` to your command line.
+
+To get information about providers for a specific SPI, append `--spi <spi>` to your command line.
+```
+
+To get more information about each SPI, it should be possible to run `kc.[sh|bat] list-spi`: 
+
+```
+Listing available SPI (see --help for more options)
```

Comment:
I'm not sure if this makes sense at all without the information provided by --full. Can you clarify the use case of just running `kc.sh list-spi`?